app/src/main/java/net/squanchy/search/SearchActivity.kt

```diff
@@ -0,0 +1,238 @@
+package net.squanchy.search
+
+import android.app.Activity
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.os.Handler
+import android.os.ResultReceiver
+import android.speech.RecognizerIntent
+import android.support.annotation.DrawableRes
+import android.support.v4.content.res.ResourcesCompat
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.Toolbar
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.widget.TextView
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import kotlinx.android.synthetic.main.activity_search.*
+import net.squanchy.R
+import net.squanchy.navigation.Navigator
+import net.squanchy.schedule.domain.view.Event
+import net.squanchy.search.view.SearchRecyclerView
+import net.squanchy.speaker.domain.view.Speaker
+import net.squanchy.support.view.enableLightNavigationBar
+import timber.log.Timber
+import java.util.concurrent.TimeUnit
+
+// Might use some refactoring later on to extract some collaborator
+class SearchActivity : AppCompatActivity(), SearchRecyclerView.OnSearchResultClickListener {
+
+    private val subscriptions = CompositeDisposable()
+
+    private lateinit var navigator: Navigator
+    private lateinit var searchService: SearchService
+
+    private var searchTextWatcher: SearchTextWatcher? = null
+
+    private var hasQuery: Boolean = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_search)
+        enableLightNavigationBar(this)
+        setupToolbar()
+
+        with(searchComponent(this)) {
+            searchService = service()
+            navigator = navigator()
+        }
+    }
+
+    private fun setupToolbar() {
+        val toolbar = findViewById<Toolbar>(R.id.search_toolbar)
+        setSupportActionBar(toolbar)
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        val querySubject = PublishSubject.create<String>()
```

Comment:
Problem is that we remove the listener in the `onStop` and also I would need to make `querySubject` a global variable, since we need it to generate the observable in the `onStart`

Reply:
Ok, nevermind then